PostFreeSurfer/PostFreeSurferPipeline.sh

```diff
@@ -70,6 +70,7 @@ opts_AddOptional '--inflatescale' 'InflateExtraScale' 'number' "surface inflatio
 opts_AddOptional '--processing-mode' 'ProcessingMode' 'HCPStyleData|LegacyStyleData' "disable some HCP preprocessing requirements to allow processing of data that doesn't meet HCP acquisition guidelines - don't use this if you don't need to" 'HCPStyleData'
 opts_AddOptional '--structural-qc' 'QCMode' 'yes|no|only' "whether to run structural QC, default 'yes'" 'yes'
 opts_AddOptional '--use-ind-mean' 'UseIndMean' 'YES or NO' "whether to use the mean of the subject's myelin map as reference map's myelin map mean, defaults to 'YES'" 'YES'
+opts_AddOptional '--metric-regression' 'MetricReg' 'OLD, NEW or BOTH' "whether to use the updated curvature-thickness regression, defaults to 'B=BOTH'" 'B'
```

Comment:
Why do we want an option to not generate one of these?  That would mean any later pipelines that use thickness and don't want curvature effects would have to be adapted to what the user chose for this setting.

I would think we want both to always be generated, without giving the user a choice to make their dataset more complicated for later handling.

Reply:
The old method is deprecated because it is substantially inferior.  We are not using corrthickness much in pipelines (I think only the IDP pipeline) other than resampling it around.

Reply:
The old version is easy to compute and not a large file.  I don't see an obvious downside to keeping backward compatibility.  I assume we don't want to use the old filename for the new output.  What exactly are you suggesting by "deprecated"?

Reply:
It's purpose is entirely superseded by the new version.

Reply:
Okay, and because of that...what do you want the pipelines to *do* about it?

Reply:
In a new study, I suspect one would only generate the new version.

Reply:
Not generating the old version is adding a difference of a "missing" file to the dataset based on when it was processed.  What is the advantage of that situation?  What if external users have been using that file, and want to replicate the methods of their old study on a different dataset?

I see more downsides than advantages to not generating the old version.  I guess the main downside I see is possible user confusion about what file to use.

Reply:
Regardless, I'm not sure there should be a way for the user to not run the new method (is the worry just that they don't have python set up for it?).  If we decide we really don't want people to run the old method by default, an option to turn it back on might be reasonable (so they don't have to resort to an older pipelines release), but that can be just a yes/no option.

Reply:
@coalsont makes a good point IMO about continuing to generate the "old" version.  Why not just automatically generate both?

Reply:
That is the current default.  We may change it in the future, as once this is available and published we will discourage use of the old method.  I think the options as defined currently are fine.

Reply:
@mharms the default at present is to generate both (though Matt seems to want to change that).  I am going further than "automatically" generating both, to "always" - why do we even want to give the user a friendly option to have "missing" files in their outputs?

Reply:
corrThickness isn't hardcoded anywhere as far as I know.  If it is, that will need to change because we would want to use this instead.  I think this discussion has run its course.

Reply:
We don't have a conclusion, so "run its course" doesn't work.  I think removing the option entirely and always generating both files results in the fewest edge cases for us to support.

Reply:
We are not doing that.  We will have a transitional period with both as a default.

Reply:
Based on what logic?  "Just do it my way" isn't very constructive.

In particular, why do we want the user to have the option to NOT generate the new version?

Reply:
That is the current behavior, so we are providing a way to support that.  We need to be able to make changes to already written pipelines.  We can have transitional periods for deprecated features.  corrThickness is obsolete with this new version.  I don't really understand why this is so controversial.  The pipelines need to be able to evolve with time.

Reply:
Huh?  I'm not saying the pipelines shouldn't change, where did you get that?  It is controversial because you aren't providing a reason for your strong position.

Why does **the user** need an option to **not run** the new method?  It doesn't overwrite a previously-released filename with different contents, right?  So the new files can't interfere with any old code (unless the user happened to already generate files with those new names), right?  If so, then why do we want an "old behavior" mode, when the new behavior doesn't change any files that the previous version generated?  Why would a user want that?

Reply:
Is the concern that someone won't like the added runtime from the new version?

Reply:
It is quite a bit longer.

Reply:
People will generally only run PostFS after they have run FreeSurfer.  How does it compare to the runtime of FreeSurfer?